Kickstarter-iOS/ViewModels/RootViewModel.swift

```diff
@@ -244,9 +250,62 @@ internal final class RootViewModel: RootViewModelType, RootViewModelInputs, Root
       .skipNil()
       .map { $0 as RootViewControllerIndex }
 
-    self.setBadgeValueAtIndex = activityViewControllerIndex
-      .takeWhen(self.viewDidLoadProperty.signal)
-      .map { index in (activitiesBadgeValue(), index) }
+    let lifecycleEvents = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.applicationWillEnterForegroundSignal
+    )
+
+    let updateBadgeValueOnLifecycleEvents = activityViewControllerIndex
+      .takeWhen(lifecycleEvents)
+      .map { index in
+        (activitiesBadgeValue(with: AppEnvironment.current.application.applicationIconBadgeNumber), index)
+      }
+
+    let selectedIndexAndActivityViewControllerIndex = Signal.combineLatest(
+      self.selectedIndex,
+      activityViewControllerIndex
+    )
+
+    let clearBadgeValueOnActivitiesTabSelected = selectedIndexAndActivityViewControllerIndex
+      .filter(==)
```

Comment:
I wonder if it might be a good idea to add an additional check here that first determines whether or not there actually *is* a badge on the activities tab before sending off the request to clear it? I think we should try to reduce the number of network requests as much as possible.

Reply:
We now only clear this if we know we have a badge value.